ffi/hello_world/c/Makefile

```diff
@@ -7,5 +7,8 @@ hello.o: hello.c
 dylib: hello.o
 	gcc -dynamiclib -undefined suppress -flat_namespace hello.o -o ../hello_world.dylib
 
+dll: hello.o
+	gcc -shared hello.o -o ../hello_world.dylib
```

Comment:
Where does GCC come from? Shouldn't we expect a Visual Studio installation?

Reply:
Yes, that is true. And it needs to be in the `PATH` as well along with `make`.
I made this change assuming they're all installed and configured properly.

Reply:
Ok, then could you please add instructions for installing GCC on Windows?

Reply:
I think `cl.exe` is the default when the command-line tools are installed:
https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019

Reply:
To use `cl.exe` from a command [this](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019#developer_command_file_locations) needs to be setup.

And there's no `make` on windows if `cl.exe` is considered the standard way.

`nmake` exists for windows.

And this makefile (say `Makefile.win`)
```makefile
hello.exe: hello.obj
	cl hello.obj

hello.obj: hello.c
	cl /c hello.c

dll: hello.obj
	link /DLL /out:..\hello_world.dll hello.obj

clean:
	rm hello.obj ..\hello_world.dll
```
`nmake -f Makefile.win dll`
Will create a `hello_world.dll` successfully but the function lookup fails with this dll.

Reply:
I must be missing something. How was the `dart:ffi` tested on windows?

Reply:
And the easiest way to set up `gcc` on windows is to use [`msys2`](https://www.msys2.org/). It has the instructions on how to set it up. If this is acceptable i.e. using `gcc` instead of `cl` then I guess that resource can be mentioned in the readme.

Reply:
I'm fine with using `gcc` in this sample if the Readme is updated.

I believe we use the Visual Studio toolchain for building the Dart SDK on Windows, which is also the toolchain we compile DLLs with for testing the FFI.

Reply:
I have made both the suggested changes.

Reply:
The dll needs to be compiled with all the necessary functions exported. You can read the updated readme on my fork and let me know if it can be truncated. @sjindel-google

Reply:
LGTM